lua/apisix/schema_def.lua

```diff
@@ -236,7 +236,7 @@ local upstream_schema = {
         key = {
             description = "the key of chash for dynamic load balancing",
             type = "string",
-            enum = {"remote_addr"},
+            pattern = [[^((uri|server_name|server_addr|request_uri|remote_port|remote_addr|query_string|host|hostname)|arg_[0-9a-zA-z_-]+)$]],
```

Comment:
code style: Single line code should be less than 80 characters long

Reply:
got it